rules/watcher.go

```diff
@@ -70,7 +70,7 @@ func (w *watcher) singleRun() {
 		time.Sleep(delay) // TODO ideally a context should be used for fast shutdown, e.g. select { case <-ctx.Done(); case <-time.After(delay) }
 	}
 	w.logger.Debug("Calling process key", zap.String("key", key))
-	w.kp.processKey(key, value, w.api, w.logger, map[string]string{}, incRuleProcessedCount)
+	w.kp.processKey(key, value, w.api, w.logger, map[string]string{"source": "watcher"}, incRuleProcessedCount)
```

Comment:
Crawler source is already being done here: https://github.com/IBM-Cloud/go-etcd-rules/blob/master/rules/int_crawler.go#L200